scipy/sparse/linalg/eigen/tests/test_svds.py

```diff
@@ -1,8 +1,10 @@
+import re
```

Comment:
I ended up basically rewriting all this. There were just a lot of holes, and it didn't make sense to leave the existing tests and add new ones that were more comprehensive.